[Bug] bin/pulsar-client NoClassDefFoundError: io/netty/incubator/channel/uring/IOUringEventLoopGroup

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Version

I built `bin/pulsar-client` using a recent version of master. Here is the commit sha 080213bdf7a00dda8f8e7c5927d83d7e6a9c5c6a.

### Minimal reproduce step

Build the pulsar project's `master` branch using `mvn -Pcore-modules,-main -T 1C clean install -DskipTests -Dspotbugs.skip=true`. I was at 080213bdf7a00dda8f8e7c5927d83d7e6a9c5c6a. I used:

```
$ java -version
openjdk version "17.0.4.1" 2022-08-12
OpenJDK Runtime Environment Temurin-17.0.4.1+1 (build 17.0.4.1+1)
OpenJDK 64-Bit Server VM Temurin-17.0.4.1+1 (build 17.0.4.1+1, mixed mode, sharing)
```

```
$ bin/pulsar-client produce -m test test
2022-11-17T22:30:47,216-0600 [main] INFO  org.apache.pulsar.client.cli.PulsarClientTool - 0 messages successfully produced
Exception in thread "main" java.lang.NoClassDefFoundError: io/netty/incubator/channel/uring/IOUringEventLoopGroup
        at org.apache.pulsar.common.util.netty.EventLoopUtil.getClientSocketChannelClass(EventLoopUtil.java:106)
        at org.apache.pulsar.client.impl.ConnectionPool.<init>(ConnectionPool.java:111)
        at org.apache.pulsar.client.impl.ConnectionPool.<init>(ConnectionPool.java:95)
        at org.apache.pulsar.client.impl.ConnectionPool.<init>(ConnectionPool.java:90)
        at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:197)
        at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:153)
        at org.apache.pulsar.client.impl.ClientBuilderImpl.build(ClientBuilderImpl.java:63)
        at org.apache.pulsar.client.cli.CmdProduce.publish(CmdProduce.java:247)
        at org.apache.pulsar.client.cli.CmdProduce.run(CmdProduce.java:239)
        at org.apache.pulsar.client.cli.PulsarClientTool.run(PulsarClientTool.java:231)
        at org.apache.pulsar.client.cli.PulsarClientTool.main(PulsarClientTool.java:271)
Caused by: java.lang.ClassNotFoundException: io.netty.incubator.channel.uring.IOUringEventLoopGroup
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
        ... 11 more
```

### What did you expect to see?

It should not have failed.

### What did you see instead?

A missing class.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

This is a bug. I already reproduce it on my computer.

```
mvn clean install -Pcore-modules,-main -DskipTests -Dspotbugs.skip
./bin/pulsar-client produce -m "message" test-topic-1
```
log:
```
❯ ./bin/pulsar-client produce -m "message" test-topic-1
2022-11-21T20:56:31,582+0800 [main] INFO  org.apache.pulsar.client.cli.PulsarClientTool - 0 messages successfully produced
Exception in thread "main" java.lang.NoClassDefFoundError: io/netty/incubator/channel/uring/IOUringEventLoopGroup
        at org.apache.pulsar.common.util.netty.EventLoopUtil.getClientSocketChannelClass(EventLoopUtil.java:106)
        at org.apache.pulsar.client.impl.ConnectionPool.<init>(ConnectionPool.java:111)
        at org.apache.pulsar.client.impl.ConnectionPool.<init>(ConnectionPool.java:95)
        at org.apache.pulsar.client.impl.ConnectionPool.<init>(ConnectionPool.java:90)
        at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:197)
        at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:153)
        at org.apache.pulsar.client.impl.ClientBuilderImpl.build(ClientBuilderImpl.java:63)
        at org.apache.pulsar.client.cli.CmdProduce.publish(CmdProduce.java:247)
        at org.apache.pulsar.client.cli.CmdProduce.run(CmdProduce.java:239)
        at org.apache.pulsar.client.cli.PulsarClientTool.run(PulsarClientTool.java:231)
        at org.apache.pulsar.client.cli.PulsarClientTool.main(PulsarClientTool.java:271)
Caused by: java.lang.ClassNotFoundException: io.netty.incubator.channel.uring.IOUringEventLoopGroup
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
        ... 11 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -647,6 +647,11 @@ flexible messaging model and an intuitive client API.</description>
 
       <dependency>
         <groupId>io.netty.incubator</groupId>
+        <artifactId>netty-incubator-transport-classes-io_uring</artifactId>
+        <version>${netty-iouring.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>io.netty.incubator</groupId>
         <artifactId>netty-incubator-transport-native-io_uring</artifactId>
         <version>${netty-iouring.version}</version>
       </dependency>

--- a/pulsar-common/pom.xml
+++ b/pulsar-common/pom.xml
@@ -133,6 +133,11 @@
 
     <dependency>
       <groupId>io.netty.incubator</groupId>
+      <artifactId>netty-incubator-transport-classes-io_uring</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>io.netty.incubator</groupId>
       <artifactId>netty-incubator-transport-native-io_uring</artifactId>
       <classifier>linux-x86_64</classifier>
     </dependency>

